### PR TITLE
[INTEG-388][integ-wiki] Space name doesn't update after moving space …

### PR DESCRIPTION
**Pull request by AtiaZ**
…wiki page

### CODE REVIEW

File: integ-wiki/integ-wiki-social/src/main/java/org/exoplatform/wiki/ext/impl/WikiSpaceActivityPublisher.java [hunk 1]
```diff
@@ -105,6 +105,12 @@ private ExoSocialActivity generateActivity(Identity ownerStream,
       }
       activity = createNewActivity(ownerIdentity.getId());
     }
+    if (PageUpdateType.MOVE_PAGE.equals(activityType)) {
+      // Delete activity of old location
+      activityManager.deleteActivity(page.getActivityId());
+      // Add new activity of new location
+      activity = createNewActivity(ownerIdentity.getId());
+    }
```
Comment (boubaker) [inline integ-wiki/integ-wiki-social/src/main/java/org/exoplatform/wiki/ext/impl/WikiSpaceActivityPublisher.java]: Should just set the new stream owner instead of delete/recreate the activity each time we move the page (I 'm not sure but it seems like it recreate the activity even if we move the page to a parent page in the same space)

File: integ-wiki/integ-wiki-social/src/main/java/org/exoplatform/wiki/ext/impl/WikiSpaceActivityPublisher.java [hunk 2]
```diff
@@ -383,16 +386,16 @@ private void saveActivity(String wikiType, String wikiOwner, String pageId, Page
       }
       return;
     }
-    
+
```
Comment (boubaker) [inline integ-wiki/integ-wiki-social/src/main/java/org/exoplatform/wiki/ext/impl/WikiSpaceActivityPublisher.java]: Should avoid this kind of changes

File: integ-wiki/integ-wiki-social/src/main/java/org/exoplatform/wiki/ext/impl/WikiSpaceActivityPublisher.java [hunk 3]
```diff
@@ -404,30 +407,29 @@ private void saveActivity(String wikiType, String wikiOwner, String pageId, Page
       try {
         space = spaceService.getSpaceByGroupId(wikiOwner);
         if (space != null) {
-          if (!isPublicInSpace(page, space)) return;
```
Comment (boubaker) [inline integ-wiki/integ-wiki-social/src/main/java/org/exoplatform/wiki/ext/impl/WikiSpaceActivityPublisher.java]: Why removing this test. In fact, we don't manage the wiki activities in a non public space

File: integ-wiki/integ-wiki-social/src/main/java/org/exoplatform/wiki/ext/impl/WikiSpaceActivityPublisher.java (hunk 1 above)
Comment (AtiaZ) [inline integ-wiki/integ-wiki-social/src/main/java/org/exoplatform/wiki/ext/impl/WikiSpaceActivityPublisher.java]: done

File: integ-wiki/integ-wiki-social/src/main/java/org/exoplatform/wiki/ext/impl/WikiSpaceActivityPublisher.java (hunk 2 above)
Comment (AtiaZ) [inline integ-wiki/integ-wiki-social/src/main/java/org/exoplatform/wiki/ext/impl/WikiSpaceActivityPublisher.java]: done

File: integ-wiki/integ-wiki-social/src/main/java/org/exoplatform/wiki/ext/impl/WikiSpaceActivityPublisher.java (hunk 3 above)
Comment (AtiaZ) [inline integ-wiki/integ-wiki-social/src/main/java/org/exoplatform/wiki/ext/impl/WikiSpaceActivityPublisher.java]: Reverted

File: integ-wiki/integ-wiki-social/src/main/java/org/exoplatform/wiki/ext/impl/WikiSpaceActivityPublisher.java [hunk 4]
```diff
@@ -361,11 +362,10 @@ private boolean isPublic(Page page) throws WikiException {
    */
   private boolean isPublicInSpace(Page page, Space space) throws WikiException {
     List<PermissionEntry> pagePermissions = page.getPermissions();
-    String groupMemberShip = MembershipEntry.ANY_TYPE + ":" + space.getGroupId();
     boolean isPublic = false;
     if(pagePermissions != null) {
       for(PermissionEntry permissionEntry : pagePermissions) {
-        if(permissionEntry.getIdType().equals(IDType.MEMBERSHIP) && permissionEntry.getId().equals(groupMemberShip)) {
+        if(permissionEntry.getIdType().equals(IDType.MEMBERSHIP)) {
```
Comment (thomasdelhomenie) [inline integ-wiki/integ-wiki-social/src/main/java/org/exoplatform/wiki/ext/impl/WikiSpaceActivityPublisher.java]: Can you explain this change please ?
What I understand is that if the page contains a membership permission (for example member:/spaces/space_1 or editor:/spaces/space_1), it is considered as public in space, which is not correct.

File: integ-wiki/integ-wiki-social/src/main/java/org/exoplatform/wiki/ext/impl/WikiSpaceActivityPublisher.java [hunk 5]
```diff
@@ -405,9 +405,9 @@ private void saveActivity(String wikiType, String wikiOwner, String pageId, Page
         space = spaceService.getSpaceByGroupId(wikiOwner);
         if (space != null) {
           if (!isPublicInSpace(page, space)) return;
-          ownerStream = identityM.getOrCreateIdentity(SpaceIdentityProvider.NAME, space.getPrettyName(), false);
           spaceUrl = space.getUrl();
           spaceName = space.getDisplayName();
+          ownerStream = identityM.getOrCreateIdentity(SpaceIdentityProvider.NAME, space.getPrettyName(), false);
```
Comment (thomasdelhomenie) [inline integ-wiki/integ-wiki-social/src/main/java/org/exoplatform/wiki/ext/impl/WikiSpaceActivityPublisher.java]: Is this change necessary ?

Comment (AtiaZ) [inline integ-wiki/integ-wiki-social/src/main/java/org/exoplatform/wiki/ext/impl/WikiSpaceActivityPublisher.java]: change reverted.

File: integ-wiki/integ-wiki-social/src/main/java/org/exoplatform/wiki/ext/impl/WikiSpaceActivityPublisher.java (hunk 4 above)
Comment (AtiaZ) [inline integ-wiki/integ-wiki-social/src/main/java/org/exoplatform/wiki/ext/impl/WikiSpaceActivityPublisher.java]: It checks if the wiki page can be read by a user who has the indicated membership permission or not. If yes, the page is considered as public in the space.

Comment (thomasdelhomenie) [inline integ-wiki/integ-wiki-social/src/main/java/org/exoplatform/wiki/ext/impl/WikiSpaceActivityPublisher.java]: So it changes the purpose of the method. Initially this method is intended to check if a page is public in a space, as described in the comments : "Check If a page can be read by all users of a space". Now it only check if the user can access it.

File: integ-wiki/integ-wiki-social/src/test/java/org/exoplatform/wiki/ext/impl/WikiSpaceActivityPublisherTest.java [hunk 6]
```diff
@@ -14,7 +15,8 @@
   public void shouldNotCreateActivityWhenUpdateTypeIsNull() throws Exception {
     // Given
     WikiService wikiService = Mockito.mock(WikiService.class);
-    WikiSpaceActivityPublisher wikiSpaceActivityPublisher = new WikiSpaceActivityPublisher(wikiService);
```
Comment (boubaker) [inline integ-wiki/integ-wiki-social/src/test/java/org/exoplatform/wiki/ext/impl/WikiSpaceActivityPublisherTest.java]: Please implement the test case that is fixed by bug. The test should fail without PR and succeed when applying PR.
